vowpalwabbit/parse_slates_example_json.h

```diff
@@ -148,65 +148,106 @@ void handle_features_value(const char* key_namespace, const Value& value, exampl
 
 template <bool audit>
 void parse_context(const Value& context, vw& all, v_array<example*>& examples, VW::example_factory_t example_factory,
-    void* ex_factory_context, std::vector<example*>& slot_examples)
+    void* ex_factory_context, std::vector<example*>& slot_examples,
+    std::unordered_map<uint64_t, example*>* dedup_examples = nullptr)
 {
   std::vector<Namespace<audit>> namespaces;
   handle_features_value(" ", context, examples[0], namespaces, all);
   all.example_parser->lbl_parser.default_label(&examples[0]->l);
-  examples[0]->l.slates.type = VW::slates::example_type::shared;
+  if (context.HasMember("_slot_id"))
+  {
+    auto slot_id = context["_slot_id"].GetInt();
+    examples[0]->l.slates.slot_id = slot_id;
+    examples[0]->l.slates.type = VW::slates::example_type::action;
+  }
+  else
+  {
+    examples[0]->l.slates.type = VW::slates::example_type::shared;
+  }
 
   assert(namespaces.size() == 0);
 
-  const auto& multi = context["_multi"].GetArray();
-  for (const Value& obj : multi)
+  if (context.HasMember("_multi"))
   {
-    auto ex = &(*example_factory)(ex_factory_context);
-    all.example_parser->lbl_parser.default_label(&ex->l);
-    ex->l.slates.type = VW::slates::example_type::action;
-    examples.push_back(ex);
-    auto slot_id = obj["_slot_id"].GetInt();
-    ex->l.slates.slot_id = slot_id;
-    handle_features_value(" ", obj, ex, namespaces, all);
-    assert(namespaces.size() == 0);
+    const auto& multi = context["_multi"].GetArray();
+    if (dedup_examples && !dedup_examples->empty())
+    {
+      for (const Value& obj : multi)
+      {
+        auto dedup_id = obj["__aid"].GetUint64();
+        auto* stored_ex = (*dedup_examples)[dedup_id];
```

Comment:
We need to handle this failing to find.

Reply:
good point, I made it throw (at least for now) since that's what the json parser does on malformed json input